File: challenges/evenOddLetters/solutions/evenOddLetters.py
# Author is github id: ddk376
# Contributors : Python group in the January 17, 2017 meetup 
# Version: Python 3.5.2

def evenOddLetters(s, unique = False ):
    """Takes a string and checks the number of letter in a word are even or odd.
      If unique is set to True, it checks the number of unique letters in a word are even or odd"""
   
    if unique: s = set(s)
    if len(s) % 2 == 0:
        print("Even")
    else:
        print("Odd")

if __name__ == "__main__":
    print("Testing evenOddLetters............................")
    evenOddLetters("llama")
    evenOddLetters("Hello world", unique = True)

File: challenges/evenOddLetters/solutions/even_odd_letters.rb
def even_odd_letters?(word: raise)
	raise ArgumentError unless !word.nil?
	letters = word.gsub(/\W+/, '') #extract letters from the word given

	# &1==0 binary test to check even/odd
	(letters.chars.uniq.size)&1==0 ? (p "#{word} has even number of unique letters") : (p "#{word} has odd unique letters")
end

even_odd_letters?(word: "w") #=> odd
even_odd_letters?(word: "AA") #=> odd
even_odd_letters?(word: "AAB") #=> even
even_odd_letters?(word: "") #=> even (cz 0 is an even number)
even_odd_letters?(word: nil) #=> ArgumentError